### PR TITLE
Scheduler testing updates

### PR DESCRIPTION
**Pull request by ltseeley**
Improves existing scheduler tests and adds more.

### CODE REVIEW

**Comment by peterschwarz** [comment]
Tested locally

File: libtransact/src/scheduler/mod.rs [hunk 1]
```diff
@@ -367,74 +366,53 @@ mod tests {
         }
     }
 
-    /// Tests a simple scheduler worklfow of processing a single transaction.
+    /// Tests a simple scheduler worklfow of processing a single transaction; this tests the
```
Comment (agunde406) [inline libtransact/src/scheduler/mod.rs]: typo worklfow -> workflow

File: libtransact/src/scheduler/mod.rs [hunk 2]
```diff
@@ -440,4 +444,84 @@ mod tests {
         let result = rx.recv().expect("Failed to receive result");
         assert_eq!(result, invalid_result_from_batch(batch));
     }
+
+    /// Tests a simple scheduler worklfow of processing a single batch with three transactions.
```
Comment (agunde406) [inline libtransact/src/scheduler/mod.rs]: worklfow -> workflow

File: libtransact/src/scheduler/serial/mod.rs [hunk 3]
```diff
@@ -261,4 +261,14 @@ mod tests {
         test_scheduler_flow_with_one_transaction(&mut scheduler);
         scheduler.shutdown();
     }
+
+    #[test]
+    pub fn test_serial_scheduler_flow_with_multiple_transactions() {
```
Comment (agunde406) [inline libtransact/src/scheduler/serial/mod.rs]: Missing comment about test

File: libtransact/src/scheduler/serial/mod.rs [hunk 4]
```diff
@@ -271,4 +271,14 @@ mod tests {
         test_scheduler_flow_with_multiple_transactions(&mut scheduler);
         scheduler.shutdown();
     }
+
+    #[test]
+    pub fn test_serial_scheduler_invalid_transaction_invalidates_batch() {
```
Comment (agunde406) [inline libtransact/src/scheduler/serial/mod.rs]: Missing test comment

File: libtransact/src/scheduler/serial/mod.rs [hunk 5]
```diff
@@ -281,4 +281,14 @@ mod tests {
         test_scheduler_invalid_transaction_invalidates_batch(&mut scheduler);
         scheduler.shutdown();
     }
+
+    #[test]
+    pub fn test_serial_scheduler_unexpected_notification() {
```
Comment (agunde406) [inline libtransact/src/scheduler/serial/mod.rs]: Missing test comment

File: libtransact/src/scheduler/mod.rs (hunk 1 above)
Comment (ltseeley) [reply libtransact/src/scheduler/mod.rs]: fixed

File: libtransact/src/scheduler/mod.rs (hunk 2 above)
Comment (ltseeley) [reply libtransact/src/scheduler/mod.rs]: fixed

File: libtransact/src/scheduler/serial/mod.rs (hunk 3 above)
Comment (ltseeley) [reply libtransact/src/scheduler/serial/mod.rs]: Added comment

File: libtransact/src/scheduler/serial/mod.rs (hunk 4 above)
Comment (ltseeley) [reply libtransact/src/scheduler/serial/mod.rs]: Added comment

File: libtransact/src/scheduler/serial/mod.rs (hunk 5 above)
Comment (ltseeley) [reply libtransact/src/scheduler/serial/mod.rs]: Added comment

**Pull request merged**